src/ThreeEditor/js/commands/DuplicateObjectCommand.ts

```diff
@@ -0,0 +1,106 @@
+import { ObjectManagementFactory } from '../../commands/factories/ObjectManagementFactory';
+import { SimulationSceneChild } from '../../Simulation/Base/SimulationContainer';
+import { SimulationElement } from '../../Simulation/Base/SimulationElement';
+import { isSimulationZone } from '../../Simulation/Base/SimulationZone';
+import { isDetector } from '../../Simulation/Detectors/Detector';
+import { isBasicFigure } from '../../Simulation/Figures/BasicFigures';
+import { isScoringFilter } from '../../Simulation/Scoring/ScoringFilter';
+import { isOutput } from '../../Simulation/Scoring/ScoringOutput';
+import { isQuantity } from '../../Simulation/Scoring/ScoringQuantity';
+import { Command, CommandJSON } from '../Command';
+import { YaptideEditor } from '../YaptideEditor';
+
+interface DuplicateObjectCommandJSON extends CommandJSON {
+	objectUuid: string;
+	actionCommands: CommandJSON[];
+}
+
+export const canBeDuplicated = (object: unknown) => {
+	return (
+		isBasicFigure(object) ||
+		isOutput(object) ||
+		isQuantity(object) ||
+		isScoringFilter(object) ||
+		isDetector(object) ||
+		isSimulationZone(object)
+	);
+};
+
+export const getDuplicateCommand = (editor: YaptideEditor, object: SimulationElement) => {
+	const commandFactory = new ObjectManagementFactory(editor);
+
+	const clone = object.duplicate();
+
+	if (isBasicFigure(clone)) {
+		return commandFactory.createDuplicateCommand('figure', clone, editor.figureManager);
+	} else if (isSimulationZone(clone)) {
+		return commandFactory.createDuplicateCommand('zone', clone, editor.zoneManager);
+	} else if (isDetector(clone)) {
+		return commandFactory.createDuplicateCommand('detector', clone, editor.detectorManager);
+	} else if (isOutput(clone)) {
+		return commandFactory.createDuplicateCommand('output', clone, editor.scoringManager);
+	} else if (isQuantity(clone) && object.parent) {
+		const output = object.parent.parent;
+
+		if (isOutput(output))
+			return commandFactory.createDuplicateCommand('quantity', clone, output);
+	} else if (isScoringFilter(clone)) {
+		return commandFactory.createDuplicateCommand('filter', clone, editor.scoringManager);
+	}
+
+	throw new Error('Object cannot be duplicated');
+};
+
+/**
+ * @param editor YaptideEditor
+ * @param object SimulationSceneChild
+ * @constructor
+ */
+export class DuplicateObjectCommand extends Command {
```

Comment:
![image](https://github.com/yaptide/ui/assets/26445545/7ab349cf-a157-4bf0-9992-300b6e5de3ab)
Trash it 🗑️ or deprecate it?